### ISSUE TITLE
Cannot run multipass as a non-admin user on macOS or Linux

### ISSUE THREAD

**Issue by rcarmo** [canonical/multipass#1437]
**Describe the bug**
Describe what your problem is.

**To Reproduce**
How, and what happened?
1. Setup Multipass as a non-admin user (using standard privilege elevation to run the installer)
2. Try to find the menu bar item - no menu bar item present on my account. Menu item appears only when logged in as an administrator.
3. Try to launch a `multipass` instance from terminal. Under my account, it errors out:

```
% multipass info --all
info failed: multipass socket access denied
Please check that you have read/write permissions to '/var/run/multipass_socket'
```

**Expected behavior**
What did you expect to happen?

I expected to be able to run the Mac GUI as a non-admin user and also to be able to issue CLI commands as a non-admin user. 

**Logs**
Please provide logs from the daemon, see [accessing logs](https://multipass.run/docs/accessing-logs) on where to find them on your platform.

**Additional info**
 - OS: `macOS Mojave 10.14.6` (also tested on Catalina, but this is my primary machine)
- `multipass  1.1.0+mac`
- `multipass info --all`

```
info failed: multipass socket access denied
Please check that you have read/write permissions to '/var/run/multipass_socket'
```

Also:

```
% ls -al /var/run/multipass_socket 
srw-rw----  1 root  admin  0 Mar 18 18:40 /var/run/multipass_socket
% id         
uid=503(rcarmo) gid=20(staff) groups=20(staff),12(everyone),61(localaccounts),399(com.apple.access_ssh),702(com.apple.sharepoint.group.1),701(com.apple.sharepoint.group.2),100(_lpoperator)
```

**Additional context**

Please bear in mind that it is a common security practice for experienced Mac users to _not_ run _anything_ on their desktop (or the CLI) as a user with administration privileges. 

Docker, for instance, works fine in this scenario - I can use both the GUI and the CLI as a non-admin user.

Adding myself to the `admin` group is not a suitable (or desirable) workaround, nor is using `sudo` (my personal user cannot do so to avoid scripting attacks).

**Comment by Saviq** [comment]
Hi @rcarmo the reason for this is that when someone gains access to the Multipass socket, they get administrative access to all files on the system, because our daemon (by design and necessity) runs as `root`. If we left it open, everyone would be able to `multipass mount / instance:rootfs` and then access all those files inside the instance.

We have a plan to improve this by adding an authentication/authorization layer on the encrypted socket connection (based on client certificates and a pre-shared passphrase for registration), at which point we'll be able to open the socket for the world.

**Comment by rcarmo** [comment]
I understand the technical aspects (and figured as much from inspection of the app). It's just that until that design limitation is remedied it a) makes it impossible for people in my situation to use Multipass at all for the foreseeable future and b) can actually be considered a security risk and further prevent adoption.

**Comment by pixlwave** [comment]
I thought I'd add that having just discovered multipass (It looks absolutely amazing by the way!), and noticing the same thing, I tried moving everything from `/Library/...` to `~/Library...` and updating the LaunchDaemon plist just to see what would happen. Running multipassd as a regular user throws the following error:
```
[2020-04-26T12:43:15.206] [error] [daemon] Caught an unhandled exception: Unable to read settings: access error (consider running with an administrative role)
```
I'm guessing that this would be related to #566 and placing things like the settings file in standard user locations too rather than hiding them in `/var/root`?

**Comment by Saviq** [comment]
@pixlwave Multipass service / daemon has to run as root to be able to access certain system facilities (at the very least create virtual network devices etc.). There's just no way to run the service as non-root. Running `multipass` the command line client without being a member of `sudo`, `admin` or some other administrative group is something we'll work on, soon.

**Comment by pixlwave** [comment]
Ahhh that’s a shame, I was hoping for something that could run entirely without root. Thanks for replying anyway.

**Comment by jenisys** [comment]
But can the **UNIX multipass-socket** above not be created with another **group=multipass** (or: multipass_users) instead of **group=admin**?

You only need read/write permission on the group of the multipass-socket, like currently any user in group=admin.
This would allow to go through authentication, meaning a user needs to be assigned to this special multipass user-group before he/she can use the socket. On installation of multipass, this user-group is created and the installing user is automatically added to this group (for example) or added on request. Other user-accounts on macOS could easily be added to the Multipass-group by an admin.

**Comment by rcarmo** [comment]
I would love to see something like this, really.

**Comment by fazalmajid** [comment]
Same here. I always run my Mac and Ubuntu accounts as non-admin non-privileged, that's pretty standard practice in the UNIX world. Very few Mac apps fail to handle this gracefully, and Ubuntu also handles this properly, e.g. in the Ubuntu Software app store.

**Comment by jenisys** [comment]
WORK-AROUND: Until it is fixed by the multipass team.

* Create a "multipass_users" group (name may very for you) on macOS / Linux
* Add your user to the `multipass_users` group
* Perform `sudo chgrp multipass_users /var/run/multipass_socket=`

NOTES:
* You may need to repeat the last step each after you have powered up your computer (and/or the multipass service was started again that creates the multipass socket).
* Proof-of-concept recipe

**Comment by rcarmo** [comment]
Any news on this? The workaround isn't really feasible for me, and it's been a year now since I filed this...

**Comment by Saviq** [comment]
Hey all, we don't want to rely on a `multipass` group, but if it would work for you if the group was used _if_ it already existed, we could add that for our next release.

**Comment by ajblendo** [comment]
Hi there, i am also experiencing the same problem here.

When trying your WORK-AROUND

> WORK-AROUND: Until it is fixed by the multipass team.
> 
> * Create a "multipass_users" group (name may vary for you) on macOS / Linux
> * Add your user to the `multipass_users` group
> * Perform `sudo chgrp multipass_users /var/run/multipass_socket`


it says: chgrp: 
`/var/run/multipass_socket: No such file or directory`

so in my case the multipass_socket is not created, so the workaround is not working. 

Here are steps i already did to fix the "access denied multipass_socket" error:


```
brew uninstall --zap multipass 
chown -R root:wheel run
update brew
update Xcode
brew install multipass
sudo start multipass
```
still getting the error message

Any update or help on this?

**Comment by rcarmo** [comment]
I have a proposal, which is to just add a setting to allow anyone on the `staff` group to access the socket. The rationale for that would be:

* It is the standard macOS group for users
* Macs are typically not _simultaneous_ multi-user machines
* Multipass is a development tool
* Being able to create, run and delete VMs does not mean people can take over the Mac itself

Also, more things could be done:

* A big, fat disclaimer can be added to explain that VMs created by one user might be accessed by another via `multipass shell`
* Multipass could keep track of which user created which VMs and manage access accordingly (or just use standard macOS privilege escalations to allow someone to manage all VMs)

Either way, it would be very nice to have this resolved. I'd take the simplest approach just so I could actually _use_ it with the GUI/menu item on my user account...

**Comment by fazalmajid** [comment]
Right, having to `chgrp` the socket every time the machine is rebooted gets really old fast.

**Comment by ricab** [comment]
Hi all, just want to let you know that we've started working on this :) Stay tuned!

**Comment by sanderdekoning** [comment]
@pixlwave @ricab Indeed, that is also what caused my issue. Only reverting any created symlinks and reinstalling multipass solved my issue of `Please ensure multipassd is running and '/var/run/multipass_socket' is accessible` (this file simply did not exist suddenly). For anyone running into this please look at:  https://github.com/canonical/multipass/issues/566#issuecomment-992969309

**Comment by Saviq** [comment]
Hey all, with #2346 we're introducing privileged `multipass set local.passphrase=<passphrase>`, which will then allow non-admin users to register with the Multipass service using `multipass register` (the exact command name may still change).

**Comment by ricab** [comment]
:point_up:

**Comment by Mailaender** [comment]
> Unrecognized settings key: 'local.passphrase'

> Error: Unknown command or alias 'register' (try "multipass help" or "multipass aliases")

**Comment by Saviq** [comment]
@Mailaender this is not released yet, you can use the prerelease package (e.g. https://github.com/canonical/multipass/runs/5189589773) if you like.

**Comment by rcarmo** [comment]
Any timeline for a release that includes this? I'd like to use/refer to an official package.

**Comment by townsend2010** [comment]
Hey @rcarmo!

We are in the process of doing a release, but we are currently blocked on getting another fix in that is particular to Windows.  Once that is done, we'll continue the release, hopefully soon.

**Comment by AdnanHodzic** [comment]
Not sure why this issue is closed, because I just installed multipass (1.10.1+mac-Darwin) on M1 Mac and I'm still experiencing following error when I try to run i.e: `microk8s install --cpu 6 --mem 16 --disk 50`

```
Please check that you have read/write permissions to '/var/run/multipass_socket'
launch failed: multipass socket access denied
```

**Comment by pgorsira** [comment]
FYI, the new command is `multipass auth`. Note you must first issue `multipass set local.passphrase` as credentialed user to set passphrase (and then enter this same passphrase when issuing the auth command).
